### ISSUE TITLE
about train

### ISSUE THREAD

**Issue by superli6** [liupei101/DSCA#9]
Hello author, thank you for your contribution. I trained a model using 100 whole slide images (WSI) from 70 patients' pathology, and I noticed that the c-index of the test dataset dropped from 0.6 to 0.26. Is this because I have too little data and the model is overfitting? I did not follow the process of using level=1, 2 but used level=0, 1 instead. Is this overfitting, or could there be other reasons?
[training] training epoch 29, avg. batch loss: 1.96642664, loss: 0.90510726, c_index: 0.72959
[training] validation epoch 29, loss: 0.92027694, c_index: 0.90909
[training] test epoch 29, loss: 0.91841877, c_index: 0.42105

[training] training epoch 69, avg. batch loss: 1.29691067, loss: 0.31053680, c_index: 0.89467
[training] validation epoch 69, loss: 0.80728710, c_index: 0.90909
[training] test epoch 69, loss: 1.07747674, c_index: 0.26316

[training] training epoch 92, avg. batch loss: 1.15009958, loss: 0.15829915, c_index: 0.95858
[training] validation epoch 92, loss: 0.75507718, c_index: 0.90909
[training] test epoch 92, loss: 0.85490167, c_index: 0.26316

**Comment by superli6** [comment]
![屏幕截图 2024-11-21 093119](https://github.com/user-attachments/assets/2a84a140-f597-4209-92b9-eebb0c6a72f1)

**Comment by liupei101** [comment]
Thanks for your attention.

It seems not overfitting, as the overall loss on test and validation sets still decreases during the training.

From my knowledge, this case could be caused by an imbalanced data split for patients in your `data_split_1`. When your data size is very limited, like 100 patients, it is often very challenging to get a balanced data split for cross-validation. This means that the case you described could be hard to avoid. 

It is recommended to produce the data splits again using a different method. This code ([here](https://github.com/mahmoodlab/Patch-GCN/blob/master/datasets/dataset_survival.py)) could be a good reference. Please contact the author of relevant codes if you have further problems.

Hope it works for you.

**Comment by superli6** [comment]
> Thanks for your attention.
> 
> It seems not overfitting, as the overall loss on test and validation sets still decreases during the training.
> 
> From my knowledge, this case could be caused by an imbalanced data split for patients in your `data_split_1`. When your data size is very limited, like 100 patients, it is often very challenging to get a balanced data split for cross-validation. This means that the case you described could be hard to avoid.
> 
> It is recommended to produce the data splits again using a different method. This code ([here](https://github.com/mahmoodlab/Patch-GCN/blob/master/datasets/dataset_survival.py)) could be a good reference. Please contact the author of relevant codes if you have further problems.
> 
> Hope it works for you.

Thank you for your reply. When I am training using 798 brca images as the training set and 115 images as the validation set, I find that the training usually stops after no more than 50 epochs. Is this phenomenon normal? If not, could you please tell me where the potential issue might be? Thank you very much!

**Comment by liupei101** [comment]
I think it seems ok, from my understanding. 

Actually, you can check if your training is good enough by inspecting the loss curves on train & test sets.

Best

**Comment by superli6** [comment]
Thank you
